### PR TITLE
feat: profile settings changes

### PR DESCRIPTION
**Pull request by sagarnaikjuspay**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
before we had only one boolean key called `collect_shipping_details_from_wallet_connector`, now its incrased to 4 ( for refrence https://github.com/juspay/hyperswitch-control-center/issues/1210)
Screenshot 2024-08-28 at 2 17 23 PM

## How did you test it?
default the toggles will be closed, and the 
`collect_billing_details_from_wallet_connector`,`always_collect_billing_details_from_wallet_connector`,`collect_shipping_details_from_wallet_connector`,`always_collect_shipping_details_from_wallet_connector` all these values will be false, when the toggle is switched on `collect_billing_details_from_wallet_connector` will set to true by default and the option can be changed any one at a time will be selected, same applies for colloct shipping address also 

## Where to test it?

- [x] INTEG
- [x] SANDBOX
- [x] PROD

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/Developer/PaymentSettings/PaymentSettings.res [hunk 1]
```diff
@@ -217,6 +217,80 @@ module ReturnUrl = {
...
+      options->Array.forEach(option => {
+        form.change(option.key, (option.key === key)->JSON.Encode.bool)
+      })
+    }
+
+    let p2RegularTextStyle = `${HSwitchUtils.getTextClass((P1, Medium))} text-grey-700 opacity-50`
+
+    React.useEffect(() => {
+      if isSelected {
+        let value: bool = options->Array.some(option => valuesDict->getBool(option.key, false))
```
Comment (PritishBudhiraja) [inline src/screens/Developer/PaymentSettings/PaymentSettings.res:247]:
```suggestion
        let value = options->Array.some(option => valuesDict->getBool(option.key, false))
```

File: src/screens/Developer/PaymentSettings/PaymentSettings.res [hunk 2]
```diff
@@ -217,6 +217,80 @@ module ReturnUrl = {
...
+        if !value {
+          switch options->Array.get(0) {
+          | Some(name) => form.change(name.key, true->JSON.Encode.bool)
+          | _ => ()
+          }
+        }
+      } else {
+        options->Array.forEach(option => {
+          form.change(option.key, false->JSON.Encode.bool)
+        })
```
Comment (PritishBudhiraja) [inline src/screens/Developer/PaymentSettings/PaymentSettings.res:257]:
```suggestion
        options->Array.forEach(option => form.change(option.key, false->JSON.Encode.bool))
```

File: src/screens/Developer/PaymentSettings/PaymentSettings.res [hunk 3]
```diff
@@ -217,6 +217,80 @@ module ReturnUrl = {
...
+          boolCustomClass="rounded-lg"
+        />
+        <p className="!text-base !text-grey-700 font-semibold"> {title->React.string} </p>
+      </div>
+      <RenderIf condition={isSelected}>
+        <div className="mt-4">
+          {options
+          ->Array.map(option => {
+            <div
+              className="flex gap-2 mb-3 items-center cursor-pointer"
```
Comment (PritishBudhiraja) [inline src/screens/Developer/PaymentSettings/PaymentSettings.res:277]: Add key

**Review by JeevaRamu0104 — APPROVED** [review]

File: src/screens/Developer/PaymentSettings/PaymentSettings.res [hunk 4]
```diff
@@ -217,6 +217,80 @@ module ReturnUrl = {
...
+
+module CollectDetails = {
+  @react.component
+  let make = (~title, ~options: array<options>) => {
+    open LogicUtils
+    let formState: ReactFinalForm.formState = ReactFinalForm.useFormState(
+      ReactFinalForm.useFormSubscription(["values"])->Nullable.make,
+    )
+    let valuesDict = formState.values->getDictFromJsonObject
+    let initValue: bool = options->Array.some(option => valuesDict->getBool(option.key, false))
```
Comment (JeevaRamu0104) [inline src/screens/Developer/PaymentSettings/PaymentSettings.res:233]:
```rescript
let initValue = options->Array.some(option => valuesDict->getBool(option.key, false))
```

Comment (JeevaRamu0104) [inline src/screens/Developer/PaymentSettings/PaymentSettings.res:233]:
```suggestion
    let initValue = options->Array.some(option => valuesDict->getBool(option.key, false))
```

**Review by JeevaRamu0104 — APPROVED** [review]

**Pull request merged**